abicheck: include functions from buildname.py

kasanchez519 opened the pull request:
binaryaudit: override with binaryaudit from python module

includes init, abicheck, buildname, util

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

abicheck: incorporate generalized funcs from buildname.py

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

binaryaudit: overwrite buildname.py

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

abicheck: rename bbutil import to util

also remove the try-except imports since it is included in binaryaudit __init__.py

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

kasanchez519 committed:
abicheck: include functions from buildname.py

binaryaudit: override with binaryaudit from py mod

includes init, abicheck, buildname, util

incorporate generalized funcs from buildname.py

overwrite buildname.py

abicheck: rename bbutil import to util

also remove the try-except imports since it is included in binaryaudit __init__.py

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

weltling commented on classes/abicheck.bbclass:
```diff
@@ -26,47 +24,9 @@ python binary_audit_gather_abixml() {
-                    bbutil.warn("Empty dump output for '{}'".format(fn))
-                    return
+    
+    build_all_xml(adir, id)
```
Where is this imported?

Perhaps we also should check about this function/module name, as it doesn't not quite matching.

Thanks

weltling commented on classes/abicheck.bbclass:
```diff
@@ -26,47 +24,9 @@ python binary_audit_gather_abixml() {
-
-                with open(out_fn, "w") as f:
-                    f.write(out)
-                    f.close()
```
Looks like there is a refactoring bug there. The write part is lost.

Thanks

weltling reviewed (changes requested):
This is a good step in the direction, however seems another round will be needed. Please see the comments.

Thanks

kasanchez519 committed:
Merge branch 'main' into main

kasanchez519 committed:
abicheck: include open fn and replace build_all_xml

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

binaryaudit: include create_path_to_xml in util

binaryaudit: include serialize_artifacts

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

abicheck: iterate through serialize_artifacts generator

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

binaryaudit: delete buildname.py

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

binaryaudit: import os and glob

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

binaryaudit: import util.py

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

binaryaudit: import bb as bb, not util

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

binaryaudit: flake8 fixes

Signed-off-by: Karissa Sanchez <kasanchez519@gmail.com>

weltling commented:
LGTM, thanks!
